core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -235,8 +235,11 @@ public class SqlValidatorImpl implements SqlValidatorWithHints {
 
   private int nextGeneratedId;
   protected final RelDataTypeFactory typeFactory;
-
-  /** The type of dynamic parameters until a type is imposed on them. */
+  /**
+   * The type of dynamic parameters until a type is imposed on them.
+   * It is assumed that dynamic parameter is nullable.
+   * See {@link #inferUnknownTypes(RelDataType, SqlValidatorScope, SqlNode)}
```

Comment:
The fix seems right. However, I'm confused about the comment for `unknownType`, it is not only used for 'dynamic parameters' actually.

Reply:
I can remove the comment from here and add new comment near assignment operator like:
```
/**
* It is assumed that unknown type is nullable by default
*/
unknownType = typeFactory.createTypeWithNullability(typeFactory.createUnknownType(), true);
```

Reply:
The new comment looks good.